[AINFRA-435] Test Setup to Validate Android AMI (`android`)

ParaskP7 opened the pull request:
Related: #14241

> [!IMPORTANT]
> DO NOT MERGE — This PR is used to test CI scenarios only

This is a draft PR that I will use to (primary) run the new Android AMI on the `android-staging` (based on `android`), which is part of [this](https://linear.app/a8c/project/provision-new-bare-metal-ci-agents-981ef3d39744) project.

FYI: Currently this will be used to test [AINFRA-435](https://linear.app/a8c/issue/AINFRA-435) but as the project progress I'm expecting I'll continue to use this dummy PR and branch to test more parts/steps/issues of the project (in the context of Android, mostly making sure that the Android AMI still works as expected).

ParaskP7 committed:
CI: Use android-staging agent to test ami-xyz

AMI Name: android-build-image-6.36.0v1.12-test-N

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 committed:
CI: Remove pull request condition to work with rebuild this branch

FYI: Doing that will help re-running a build without from Buildkite
itself, using the 'Rebuild this branch' option, without the need to
push a new commit but still have all jobs running.

AliSoftware committed:
Add a CI step to print some debug values

AliSoftware committed:
Force CI rebuild
